src/main/java/com/minecolonies/core/entity/ai/workers/service/EntityAIWorkCook.java

```diff
@@ -132,14 +129,10 @@ public IAIState startWorking()
     @Override
     protected boolean isSmeltable(final ItemStack stack)
     {
-        //Only return true if the item isn't queued for a recipe. 
-        if (!building.getIsCooking())
-        {
-            return ItemStackUtils.ISCOOKABLE.test(stack) && !isItemStackForAssistant(stack)
-                     && !building.getModuleMatching(ItemListModule.class, m -> m.getId().equals(FOOD_EXCLUSION_LIST))
-                           .isItemInList(new ItemStorage(MinecoloniesAPIProxy.getInstance().getFurnaceRecipes().getSmeltingResult(stack)));
-        }
-        return false;
+        //Only return true if the item isn't queued for a recipe.
+        return ItemStackUtils.ISCOOKABLE.test(stack)
+                 && !building.getModuleMatching(ItemListModule.class, m -> m.getId().equals(FOOD_EXCLUSION_LIST))
```

Comment:
should be able to query the Module directly from BuildingModules instead of trying to match by class and id